contracts/integrations/test/internal-integration-tests/exchange_wrapper_test.ts

```
@@ -0,0 +1,1003 @@
+                const defaultTakerAssetAddress = deployment.tokens.erc20[1];
+                const feeToken = deployment.tokens.erc20[2];
+                const differentTakerAssetData = assetDataUtils.encodeERC20AssetData(feeToken.address);
+                const fillTestInfo: FillTestInfo[] = [];
+
+                signedOrders = [
+                    await makerActor.signOrderAsync(),
+                    await makerActor.signOrderAsync(),
+                    await makerActor.signOrderAsync({
+                        takerAssetData: differentTakerAssetData,
+                    }),
+                ];
+                const takerAssetFillAmount = Web3Wrapper.toBaseUnitAmount(new BigNumber(100000), 18);
+
+                await testMarketSellOrdersNoThrow(
+                    signedOrders,
+                    takerAssetFillAmount,
+                    DeploymentManager.protocolFee.times(signedOrders.length),
+                );
+            });
+
+            it('should fill a signedOrder that does not use the same takerAssetAddress', async () => {
+                const defaultTakerAssetAddress = deployment.tokens.erc20[1];
+                const feeToken = deployment.tokens.erc20[2];
+                const differentTakerAssetData = assetDataUtils.encodeERC20AssetData(feeToken.address);
+                const fillTestInfo: FillTestInfo[] = [];
+
+                signedOrders = [
+                    await makerActor.signOrderAsync(),
+                    await makerActor.signOrderAsync(),
+                    await makerActor.signOrderAsync({
+                        takerAssetData: differentTakerAssetData,
+                    }),
+                ];
+                const takerAssetFillAmount = Web3Wrapper.toBaseUnitAmount(new BigNumber(100000), 18);
+
+                await testMarketSellOrdersNoThrow(signedOrders, takerAssetFillAmount, constants.ZERO_AMOUNT);
+            });
+        });
```

Maybe a case where you throw in an invalid order?